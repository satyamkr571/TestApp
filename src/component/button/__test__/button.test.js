import React from "react";
import ReactDOM from "react-dom";
import Button from "../button";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
afterEach(cleanup);
it("BUTTON COMPONENT IS RENDERED", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("BUTTON COMPONENT IS RENDERED", () => {
  const { getByTestId } = render(<Button label="Click Me" />);
  expect(getByTestId("buttontestid")).toHaveTextContent("Click Me");
});

it("BUTTON COMPONENT IS RENDERED", () => {
  const { getByTestId } = render(<Button label="Submit"></Button>);
  expect(getByTestId("buttontestid")).toHaveTextContent("Submit");
});

it("it matches snapshot", () => {
  const tree = renderer.create(<Button label="Click Me"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
it("it matches snapshot", () => {
  const tree = renderer.create(<Button label="Click "></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
