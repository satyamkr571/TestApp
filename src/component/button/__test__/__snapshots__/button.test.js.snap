// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="button"
  data-testid="buttontestid"
>
  On
  Click Me
</div>
`;

exports[`it matches snapshot 2`] = `
<div
  className="button"
  data-testid="buttontestid"
>
  On
  Click 
</div>
`;
